sdk/servicebus/Azure.Messaging.ServiceBus/perf/Scenarios/AbandonMessages.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Test.Perf;
+
+namespace Azure.Messaging.ServiceBus.Perf.Scenarios
+{
+    public sealed class AbandonMessages : SettleMessagesBase
+    {
+        public AbandonMessages(SizeCountOptions options) : base(options)
+        {
+        }
+
+        protected override async Task SettleAsync(ServiceBusReceivedMessage message, CancellationToken cancellationToken)
+        {
+            await Receiver.AbandonMessageAsync(message, cancellationToken: cancellationToken);
```

Comment:
Should we consider not awaiting here and elide the state machine to avoid introducing additional overhead?

Reply:
If not, I think we probably want `ConfigureAwait(false)` here.

Reply:
/cc @pakrym what do you think?

Reply:
Funny enough, I `awaited` in the Event Hubs tests too.   Now I'm wondering if I should go back and tune that.   Probably doesn't matter, since the goal is to observe changes across release versions rather than eeking out raw performance.

Reply:
Yeah, I think the goal is to mirror what customer code might be doing.

Reply:
Do we actually need ConfigureAwait(false) in this context?

Reply:
According to Stephen Toub, it's recommended to continue using in .NET Core when the code is agnostic to synchronization context.   _([ref](https://devblogs.microsoft.com/dotnet/configureawait-faq/#ive-heard-configureawaitfalse-is-no-longer-necessary-in-net-core-true))_   I don't think it's strictly needed, but it's the recommended pattern for console applications.

Reply:
I just meant since we are controlling the context in which this code is being run, similar to our live tests where we don't use ConfigureAwait.

Reply:
Looks like most of the other perf tests are not using it:
https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/search/Azure.Search.Documents/perf/Azure.Search.Documents.Perf
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/perf/Azure.Storage.Blobs.Perf/Scenarios/DownloadSasUriBlobClient.cs

Reply:
Leaving it off won't cause correctness problems, but it may add overhead and make results less deterministic if there's contention on a synchronization context such as when we're running with higher degrees of concurrency.

That said, I've also heard guidance that says there's really no synchronization context in the `netcoreapp` targets.   I'm not 100% sure whether it is impactful to leave it off or not... but I know that including it won't hurt.  😄

Reply:
/cc @pakrym what do you think?

1. I don't think ConfigureAwait is needed here as we are not running in any special sync context.
2. I don't have a strong opinion on awaiting or not. I lean towards having the perf test looking as normal as possible (similar to how customers would use our clients).